spec/associations/has_many_nested_spec.rb

```diff
@@ -0,0 +1,323 @@
+          nested = Nested.query.params('sys.id' => '3gpECMvSLSOC8wgkQ6GQ2q', include: 3).load.first
+          child_1 = nested.children.first
+          expect(nested.nested_children_by(:name)).not_to be_empty
+          expect(nested.nested_children_by(:name)).to eq(
+            child_1.name => nil
+          )
+        }
+      end
+
+      it 'generates flattened structures to use as paths for entries' do
+        vcr('association/nested_without_root_middle_higher_include') {
+          nested = Nested.query.params('sys.id' => '3gpECMvSLSOC8wgkQ6GQ2q', include: 3).load.first
+          child_1 = nested.children.first
+          expect(nested.all_child_paths_by(:name)).not_to be_empty
+          expect(nested.all_child_paths_by(:id)).to eq [[child_1.id]]
+        }
+      end
+
+      it 'generates a path for for entries that match a value' do
+        vcr('association/nested_without_root_middle_higher_include') {
+          nested = Nested.query.params('sys.id' => '3gpECMvSLSOC8wgkQ6GQ2q', include: 3).load.first
+          child_1 = nested.children.first
+          expect(nested.find_child_path_by(:name, child_1.name)).not_to be_empty
+          expect(nested.find_child_path_by(:name, child_1.name)).to eq [[child_1.name]]
+
+          expect(nested.find_child_path_by(:name, 'foo')).to be_empty
+          expect(nested.find_child_path_by(:name, 'foo')).to eq []
+        }
+      end
+    end
+
+    context 'only has parent' do
+      it 'parent exists' do
+        vcr('association/nested_without_root_childless') {
+          nested = Nested.find('2pFubMJuVqMSisGeCmoqem')
+          expect(nested.parent).to be_a Nested
+          expect(nested.parent.name).to eq 'Nested 2'
+        }
+      end
+
+      it 'has no children' do
+        vcr('association/nested_without_root_childless') {
+          nested = Nested.find('2pFubMJuVqMSisGeCmoqem')
+
+          expect(nested.children?).to be_falsey
+          expect(nested.children).to be_a ::Array
+          expect(nested.children.size).to eq 0
+        }
+      end
+
+      it 'is not the root' do
+        vcr('association/nested_without_root_childless') {
+          nested = Nested.find('2pFubMJuVqMSisGeCmoqem')
+          expect(nested.root?).to be_falsey
+          expect(nested.root.name).to eq 'Nested 1'
+        }
+      end
+
+      it 'has ancestors' do
+        vcr('association/nested_without_root_childless') {
+          nested = Nested.find('2pFubMJuVqMSisGeCmoqem')
+          expect(nested.ancestors).not_to be_empty
+          expect(nested.ancestors.size).to eq 2
+        }
+      end
+
+      it 'find ancestors will return an enumerator of ancestors - if no block given' do
+        vcr('association/nested_without_root_child_parent') {
+          nested = Nested.find('2pFubMJuVqMSisGeCmoqem')
+          expect(nested.find_ancestors.to_a.map(&:name)).to eq ['Nested 2', 'Nested 1']
+        }
+      end
+
+      it 'find ancestos execute the block for each ancestor - if block given' do
```

Comment:
`ancestos` -> `ancients`

Again I'm wondering about a **to** between `ancestors` and `execute`.